infrastructure/functions/detect-changes-for-worker/ecs.js

```diff
@@ -0,0 +1,187 @@
+
+    // And add in the post-commands last
+    allCommands.push(...(post || []));
+
+    // Return a new array of commands with everything chained using '&&' so that execution will terminate
+    // as soon as any command in the chain fails.
+    return ['/bin/sh', '-c', allCommands.join(' && ')];
+}
+
+/**
+ * Returns the appropriate ECS Task Definition name based on the current action
+ */
+function getTaskDefinitionName(options, event) {
+    const { GDAL_TASK_DEFINITION, TIPPECANOE_TASK_DEFINITION } = options.env;
+    const { action } = event;
+
+    switch (action) {
+        case 'gdal':
+        case 'ogr2ogr':
+        case 'enrichment':
+                return GDAL_TASK_DEFINITION;
+        case 'tippecanoe':
+            return TIPPECANOE_TASK_DEFINITION;
+    }
+
+    throw new Error(`No Fargate Task Definition defined for ${action}`);
+}
+
+/**
+ * Returns the appropriate ECS Container Definition name based on the current action
+ */
+function getFargateContainerDefinitionName(options, event) {
+    const { GDAL_CONTAINER_DEFINITION, TIPPECANOE_CONTAINER_DEFINITION } = options.env;
+    const { action } = event;
+
+    switch (action) {
+        case 'gdal':
+        case 'ogr2ogr':
+        case 'enrichment':
+                return GDAL_CONTAINER_DEFINITION;
+        case 'tippecanoe':
+            return TIPPECANOE_CONTAINER_DEFINITION
+    }
+
+    throw new Error(`No Fargate Container Definition Name defined for ${action}`);
+}
+
+/**
+ * Executes a (known) container in Fargate with the provided command line parameters.
+ */
+async function executeRawCommand(options, event) {
+    const { ecs, logger } = options.deps;
+    const { env } = options;
+    const { ECS_CLUSTER, VPC_SUBNET_ID } = env;
+
+    // If there are pre- or post- commands defined, wrap up the primary command
+    const containerCommand = wrapContainerCommand(options, event.pre, event.command, event.post);
+
+    // Get the name of the container that we are using
+    const containerDefinitionName = getFargateContainerDefinitionName(options, event);
+
+    // Create the full Task parameter object and execute
+    const params = {
+        taskDefinition: getTaskDefinitionName(options, event),
+        cluster: ECS_CLUSTER,
+        launchType: 'FARGATE',
+        count: 1,
+        networkConfiguration: { // Must be specified for tasks with `awsvpc` networking and awsvpc networking is required for FARGATE launch types
+            awsvpcConfiguration: {
+                subnets: [
+                    VPC_SUBNET_ID
+                ],
+                assignPublicIp: 'DISABLED',
+                securityGroups: []
+            }
+        },
+        overrides: {
+            containerOverrides: [
+                {
+                    name: containerDefinitionName,
+                    command: containerCommand
+                }
+            ]
+        }
+    };
+
+    logger.info(`Executing ECS Task...`, JSON.stringify(params, null, 2));
```

Comment:
Where does this end up logging to? Cloudwatch?

Reply:
Yes, it goes into CloudWatch logs under the log group defined [here](https://github.com/usds/justice40-tool/pull/230/files#diff-07622d56543936edef5ffb88b736a04eafb4273a995496b591f0851f144bb8cdR127).